code/handlers/ldap.q

```diff
@@ -32,6 +43,19 @@ unblock:{[usr]
   ];
  };
 
+bind:{[sess;customDict]
+  defaultKeys:`dn`cred`mech;
+  defaultVals:```;
+  defaultDict:defaultKeys!defaultVals;
+  if[customDict~(::);customDict:()!()];
+  if[99h<>type customDict;'"customDict must be (::) or a dictionary"];
+  updDict:defaultDict,customDict;
+  bindSession:.ldap.bind_s[sess;;;]. updDict defaultKeys;
+  bindSession
+  }
+
+/authenticate:{[dict].ldap.bind[.ldap.sessionID;(enlist dict[`bind_dn])!enlist dict[`pass]]}
+
 login:{[user;pass]                                              / validate login attempt
   incache:.ldap.cache user;                                     / get user from inputs
   dict:`version`server`port`bind_dn`pass!(.ldap.version;.ldap.server;.ldap.port;.ldap.buildDN user;pass);
```

Comment:
if we modify this, it simplifies the below call to `.ldap.bind`

```suggestion
  dict:`dn`cred!(.ldap.buildDN user;pass);
```

then `dict` can be passed directly e.g. `.ldap.bind[.ldap.sessionID;dict]` instead of having to rename dictionary keys.

the other keys here aren't needed for `.ldap.bind`, so we can get rid of them